Userland/Libraries/LibTextCodec/ShiftJISConverter.h

```diff
@@ -0,0 +1,7064 @@
+/*
+ * Copyright (c) 2021, Łukasz Patron <priv.luk@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Types.h>
+
+namespace {
+// https://unicode.org/Public/MAPPINGS/OBSOLETE/EASTASIA/JIS/SHIFTJIS.TXT
+u32 convert_shift_jis_to_utf8(u16 in)
+{
+    switch (in) {
+
+#define MAP(X, Y) \
+    case X:       \
+        return Y
+
+        MAP(0x20, 0x0020);
```

Comment:
If those values are all dense (meaning basically without holes), maybe you should use an `Array` instead, since that would probably be a lot faster than the `switch` statement.

Reply:
Would a HashMap be ok here?

Reply:
Anyway, I'll see if array would work too.

Reply:
A HashMap requires a hash operation for each lookup, an array just requires an addition (base ptr + offset). Also, an `Array` can be `constexpr`. If this is less than about three or four blocks, just make that number of `Array`s, and it should be much faster, I think.

Reply:
I ran a basic python script to find gaps and it...found over 80 of them so that's a no-go.